packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -20,64 +23,413 @@ const double _kHandleSize = 22.0;
+  }
+
+  @override
+  MultiChildRenderObjectElement createElement() => MultiChildRenderObjectElement(this);
+}
+
+class _TextSelectionToolbarItemsRenderBox extends RenderBox with ContainerRenderObjectMixin<RenderBox, FlexParentData>, RenderBoxContainerDefaultsMixin<RenderBox, FlexParentData> {
+  _TextSelectionToolbarItemsRenderBox({
+    @required bool isAbove,
+    @required bool overflowOpen,
+  }) : assert(overflowOpen != null),
+       assert(isAbove != null),
+       _isAbove = isAbove,
+       _overflowOpen = overflowOpen,
+       super();
+
+  int _lastIndexThatFits = -1;
+
+  bool _isAbove;
+  bool get isAbove => _isAbove;
+  set isAbove(bool value) {
+    if (value == isAbove) {
+      return;
+    }
+    _isAbove = value;
+    markNeedsLayout();
+  }
+
+  bool _overflowOpen;
+  bool get overflowOpen => _overflowOpen;
+  set overflowOpen(bool value) {
+    if (value == overflowOpen) {
+      return;
+    }
+    _overflowOpen = value;
+    markNeedsLayout();
+  }
+
+  // Lay out all children, regardless of whether or not they will be painted or
+  // placed with an offset. Find which child overflows, if any.
+  void _layoutChildren() {
+    int i = -1;
+    double width = 0.0;
+    visitChildren((RenderObject renderObjectChild) {
+      i++;
+      final RenderBox child = renderObjectChild as RenderBox;
+      child.layout(constraints.loosen(), parentUsesSize: true);
+      width += child.size.width;
+
+      if (width > constraints.maxWidth && _lastIndexThatFits == -1) {
+        _lastIndexThatFits = i - 1;
+      }
+    });
+
+    // If the last child overflows, but only because of the width of the
+    // overflow button, then just show it and hide the overflow button.
+    final RenderBox navButton = firstChild;
+    if (_lastIndexThatFits != -1 && _lastIndexThatFits == childCount - 2
+      && width - navButton.size.width <= constraints.maxWidth) {
+      _lastIndexThatFits = -1;
+    }
+  }
+
+  // Set the offset of all of the children that will be painted.
+  void _placeChildren() {
+    int i = -1;
+    Size nextSize = const Size(0.0, 0.0);
+    double fitWidth = 0.0;
+    final RenderBox navButton = firstChild;
+    double overflowHeight = overflowOpen && !isAbove ? navButton.size.height : 0.0;
+    visitChildren((RenderObject renderObjectChild) {
+      i++;
+
+      // The navigation button is placed after iterating all children.
+      if (renderObjectChild == firstChild) {
+        return;
+      }
+
+      final RenderBox child = renderObjectChild as RenderBox;
+      final FlexParentData childParentData = child.parentData as FlexParentData;
+
+      // If the current child is not displayed, place it offscreen to avoid it
+      // interfering with hit detection.
+      if (_lastIndexThatFits != -1) {
+        if ((!overflowOpen && i > _lastIndexThatFits)
+          || (overflowOpen && i <= _lastIndexThatFits)) {
+          childParentData.offset = Offset(-child.size.width, -child.size.height);
```

Comment:
Is there any other way to make a child that isn't painted not be hit detected?

Reply:
I wonder if you can override defaultHitTestChildren to walk through the list starting with  _lastIndexThatFits.

Reply:
Yeah that would work, I'll implement it.